### PR TITLE
Add userinfo Command

### PR DESCRIPTION
**Pull request by aexolate**
This is a simple command to get information about the user. (roles, perms, ID, join date).

This is my attempt for issue #2. I have no idea why github is displaying changes in some lines when I didn't change anything above. (my code is at the bottom) Not sure if this is how it is supposed to be done or needed

### CODE REVIEW

**Comment by NaamloosDT** [comment]
Seems fine to me, I'll check when I'm home

**Comment by Emzi0767** [comment]
Nitpick: it's a better idea to check for permissions using bitwise flags instead of the string. For example:

`(perms & Permissions.Administrator) != 0`

**Comment by NaamloosDT** [comment]
@aexolate Could you try doing what emzi said? It's not a very difficult change c:Z

File: ModCore/Commands/Main.cs [hunk 1]
```diff
@@ -1,197 +1,226 @@
...
+
+            if (usr.IsBot) embed.Title += " __[BOT]__ ";
+            if (usr.IsOwner) embed.Title += " __[OWNER]__ ";
+
+            embed.Description =
+                $"Registered on     : {usr.CreationTimestamp.DateTime.ToShortDateString()} {usr.CreationTimestamp.DateTime.ToShortTimeString()}\n" +
+                $"Joined Guild on  : {usr.JoinedAt.DateTime.ToShortDateString()} {usr.JoinedAt.DateTime.ToShortTimeString()}";
+
+            string roles = String.Empty;
+            foreach (var r in usr.Roles) roles += $"[{r.Name}] ";
```
Comment (uwx) [inline ModCore/Commands/Main.cs]: Please use StringBuilder here.

File: ModCore/Commands/Main.cs [hunk 2]
```diff
@@ -1,197 +1,226 @@
...
+                .WithTitle($"@{usr.Username}#{usr.Discriminator} - ID: {usr.Id}");
+
+            if (usr.IsBot) embed.Title += " __[BOT]__ ";
+            if (usr.IsOwner) embed.Title += " __[OWNER]__ ";
+
+            embed.Description =
+                $"Registered on     : {usr.CreationTimestamp.DateTime.ToShortDateString()} {usr.CreationTimestamp.DateTime.ToShortTimeString()}\n" +
+                $"Joined Guild on  : {usr.JoinedAt.DateTime.ToShortDateString()} {usr.JoinedAt.DateTime.ToShortTimeString()}";
+
+            string roles = String.Empty;
```
Comment (uwx) [inline ModCore/Commands/Main.cs]: Use 'var' when possible.

File: ModCore/Commands/Main.cs [hunk 3]
```diff
@@ -1,197 +1,226 @@
...
+            foreach (var r in usr.Roles) roles += $"[{r.Name}] ";
+            if (roles == String.Empty) roles = "*None*";
+            embed.AddField($"Roles", roles);
+
+            string perms = usr.PermissionsIn(ctx.Channel).ToPermissionString();
```
Comment (uwx) [inline ModCore/Commands/Main.cs]: Use 'var' when possible.

File: ModCore/Commands/Main.cs [hunk 4]
```diff
@@ -1,197 +1,226 @@
...
+            if (!(perms.Contains("Read messages") || perms.Contains("Administrator"))) perms = "**[!] User can't see this channel!**\n" + perms;
```
Comment (uwx) [inline ModCore/Commands/Main.cs]: Use bitwise flags enum comparison instead.

File: ModCore/Commands/Main.cs [hunk 5]
```diff
@@ -1,197 +1,226 @@
...
+            if (perms == String.Empty) perms = "*None*";
+            embed.AddField($"Permissions", perms);
+
+            embed.WithFooter($"{ctx.Guild.Name} / #{ctx.Channel.Name} / {DateTime.Now.ToShortDateString()} {DateTime.Now.ToShortTimeString()}");
```
Comment (uwx) [inline ModCore/Commands/Main.cs]: I think you can use interpolation format string instead of `DateTime.Now.ToShortDateString()` here.

**Comment by aexolate** [comment]
Is there an equivalent for Permissions.ReadMessages? because I cant find it. I can't check if the user can actually access the text channel or is there another way round it?

File: ModCore/Commands/Main.cs [hunk 6]
```diff
@@ -1,197 +1,226 @@
...
 using ModCore.Database;
 using ModCore.Entities;
 
-namespace ModCore.Commands
-{
-    public class Main
-    {
-        private DatabaseContextBuilder Database { get; }
-
+namespace ModCore.Commands
```
Comment (uwx) [inline ModCore/Commands/Main.cs]: I'm not sure why the diff is so cluttered. Did the encoding change?

**Comment by Emzi0767** [comment]
Read messages is now `AccessChannels` to reflect how it is in the client UI now.

File: ModCore/Commands/Main.cs (hunk 1 above)
Comment (Emzi0767) [reply ModCore/Commands/Main.cs]: Or `string.Join()` and LINQ.

File: ModCore/Commands/Main.cs (hunk 6 above)
Comment (aexolate) [reply ModCore/Commands/Main.cs]: Visual Studio did warn me about encoding but I don't know what to do with it.

Thanks, I will make the changes later on

**Comment by aexolate** [comment]
Ok thanks

**Comment by aexolate** [comment]
Made the changes

**Comment by NaamloosDT** [comment]
Seems good to me.

**Pull request merged**